### PR TITLE
Implement replace first digraph

### LINKED ISSUE #850: Replace first occurance digraph
Equivalent to 05ab1e's `.;`

(created by lyxal [here](https://chat.stackexchange.com/transcript/message/60887053))

### PR DESCRIPTION
**Pull request by Lyxal**
Closes #850

### CODE REVIEW

**Review by cgccuser — APPROVED** [review]
Looks good to me, and thanks for the tests. You could maybe even remove the last two ones unless they're important edge cases

**Comment by cgccuser** [comment]
Might be useful to also have a "replace nth occurrence" digraph

**Review by AMiller42 — APPROVED** [review]
lgtm

File: documents/knowledge/elements.yaml [hunk 1]
```diff
@@ -13,30 +13,30 @@
   arity: 1
   description: Return the inverse (negation) of the truthiness of an item.
   overloads:
-    num: not a
-    str: a != "" | len(a) > 0
-    lst: a != [] | len(a) > 0
+      num: not a
```
[the remark this answers is not in the document]
Comment (Lyxal) [reply documents/knowledge/elements.yaml:16]: Because I actually enabled my linter for once. Apparently it had never been linting my YAML.

**Comment by Steffan153** [comment]
yay now i can write my vyxal solution
